### PR TITLE
Faster taxonomy term sorting

### PR DESCRIPTION
**Pull request by defunctl**

### CODE REVIEW

**Review by jbrinley — APPROVED** [review]
Neat!

**Comment by defunctl** [comment]
@jbrinley updated with caching. Please note I have logic on the other project that clears this cache when terms change, so I added a filter to control the cache length.

File: ModularContent/Fields/Post_List.php [hunk 1]
```diff
@@ -559,84 +570,127 @@ protected static function get_p2p_filtered_ids( $connection_id, $post_ids ) {
...
+	private function get_taxonomy_terms() {
+		global $_wp_using_ext_object_cache;
+
+		$transient = 'panels_tax_terms';
+
+		// Store current object cache
+		$object_cache = $_wp_using_ext_object_cache;
+
+		// Force transients to use the database
+		$_wp_using_ext_object_cache = null;
```
Comment (jbrinley) [inline ModularContent/Fields/Post_List.php:651]: This is a really great solution. I wonder if there should be a filter to determine if we allow object cache usage or always force DB, but I link I lean on the side of leaving it as-is.

File: ModularContent/Fields/Post_List.php [hunk 2]
```diff
@@ -559,84 +570,127 @@ protected static function get_p2p_filtered_ids( $connection_id, $post_ids ) {
...
+
+				$panels_terms[ $taxonomy_name ] = $options;
 			}
-			usort( $options, [ $this, 'sort_by_label' ] );
-			$blueprint[ 'taxonomies' ][ $taxonomy_name ] = $options;
+
+			// Force transients to use the database
+			$_wp_using_ext_object_cache = null;
+
+			set_transient( $transient, $panels_terms, apply_filters( 'tribe/panels/post_list/cache_length', DAY_IN_SECONDS ) );
```
Comment (jbrinley) [inline ModularContent/Fields/Post_List.php]: Can we call it `cache_expiration` instead of `cache_length`?

**Pull request merged**